src/main/java/make_comment/use_case/MakeCommentInteractor.java

```diff
@@ -0,0 +1,102 @@
+ * with the entities and ultimately send information back out into the UI.
+ */
+public class MakeCommentInteractor implements MakeCommentInputBoundary {
+    private final MakeCommentGateway dataAccess;
+    private final MakeCommentOutputBoundary presenter;
+
+    public MakeCommentInteractor(MakeCommentGateway dataAccess, MakeCommentOutputBoundary presenter) {
+        this.dataAccess = dataAccess;
+        this.presenter = presenter;
+    }
+
+    @Override
+    public void constructAndSaveCommentAndUpdatePost (MakeCommentRequestModel mCRM) {
+        try {
+            int userId = mCRM.getUserId();
+            String body = mCRM.getCommentBody();
+            int commentId = mCRM.getSelfId();
+            int postId = mCRM.getPostId();
+            constructAndSaveCommentHelper(userId, body, commentId);
+            updatePostHelper(postId, commentId);
+            MakeCommentResponseModel responseModel = new
+                    MakeCommentResponseModel(true, "");
+            this.presenter.present(responseModel);
+
+        } catch (Exception e) {
+            MakeCommentResponseModel responseModel = new
+                    MakeCommentResponseModel(false, "Comment was not created");
+            this.presenter.present(responseModel);
+        }
+    }
+    @Override
+    public int getCurrentUserID() {
+        return CurrentUser.getCurrentUserId();
+    }
+
+    @Override
+    public int getNumCommentCreated() {
+        return this.dataAccess.getNumComments();
+    }
+
+    private void updatePostHelper(int postId, int commentId){
+        List<String[]>  postData = dataAccess.getCurrentPosts();
+        for(int x = 1; x < postData.size(); x++){
+            String currentPostId = postData.get(x)[0];
+            if (Integer.parseInt(currentPostId) == postId){
+                String newRepliesString;
+                if (postData.get(x)[9].equals("")){
+                     newRepliesString =  Integer.toString(commentId);
+                } else {
+                     newRepliesString = postData.get(x)[9] + (" " + commentId);
+                }
+
+                String[] newPost = {
+                        postData.get(x)[0],
+                        postData.get(x)[1],
+                        postData.get(x)[2],
+                        postData.get(x)[3],
+                        postData.get(x)[4],
+                        postData.get(x)[5],
+                        postData.get(x)[6],
+                        postData.get(x)[7],
+                        postData.get(x)[8],
+                        newRepliesString
+                };
+                postData.remove(x);
+                postData.add(newPost);
+                break;
+
+            }
+        }
+        dataAccess.updatePostDB(postData);
+    }
+
+    private void constructAndSaveCommentHelper(int userId, String body, int commentId){
+        if (body.equals("")){
+            throw new RuntimeException();
```

Comment:
May be a better throwable than RuntimeException, or at least we should give it some kind of method